lib/Common.pm

```diff
@@ -2,32 +2,50 @@ package Common;
 
 use strict;
 use warnings;
+use warnings FATAL => 'all';
 
 use Exporter qw(import);
 
 # The below methods are used by build_java.pl and tests/all.pl; if the
 # location of Common.pm ever changes, update the includes in those files
 # to point to the new location.
-our @EXPORT_OK = qw(get_jar_files);
+our @EXPORT_OK = qw(get_jar_files detect_jar_file);
+
+sub detect_jar_file {
+    # Detect the correct path to a jar file.
+    # Die if the jar is missing.
+
+    my @locations = qw(/usr/share/java /usr/lib/java);
```

Comment:
In case in the future you main need it, it got this:
$ ls -l /usr/bin/java
/usr/bin/java -> /System/Library/Frameworks/JavaVM.framework/Versions/Current/Commands/java
/usr/bin/javac -> /System/Library/Frameworks/JavaVM.framework/Versions/Current/Commands/javac

Reply:
Hmmm and the locations of the required Jars? I'm guessing they're not shipped with Mac and we'd have to include Brew/MacPorts installation paths... Hmmmm